lib/req/steps.ex

```diff
@@ -829,4 +825,31 @@ defmodule Req.Steps do
   defp parse_netrc([], nil, acc) do
     acc
   end
+
+  defp put_redirect_location(request, location) do
+    if String.starts_with?(location, "/") do
+      url = URI.parse(location)
+      update_in(request.url, &%{&1 | path: url.path, query: url.query})
+    else
+      url = URI.parse(location)
+      put_in(request.url, url)
+    end
+  end
+
+  defp remove_credentials_if_untrusted(%{location_trusted: true} = request, _), do: request
+
+  defp remove_credentials_if_untrusted(%{url: url} = request, location) do
+    if URI.parse(location).host == url.host do
+      request
+    else
+      remove_credentials(request)
+    end
+  end
+
+  defp remove_credentials(request) do
+    headers = List.keydelete(request.headers, "authorization", 0)
+    request_steps = Enum.reject(request.request_steps, fn {_, step, _} -> step == :auth end)
+
+    struct(request, headers: headers, request_steps: request_steps)
```

Comment:
```suggestion
    %{request | headers: headers, request_steps: request_steps}
```

Reply:
I am just curious as to why one is preferred over the other? The only difference I'm aware of (I think) is that `struct/2` discards keys that aren't in the struct (which doesn't seem to matter one way or another here). Thanks for explaining!

Reply:
yeah strucr!/2 would totally work too. Don't have a specific reason, I guess it is personal preference.